### PR TITLE
Refactor mindcraft_info() to avoid sprintf().  Does part of the work …

### PR DESCRIPTION
**Pull request by backwardsEric**
…of resolving https://github.com/hengband/hengband/issues/2858 and https://github.com/hengband/hengband/issues/2859 .

### CODE REVIEW

File: src/mind/mind-info.cpp [hunk 1]
```diff
@@ -66,179 +50,109 @@ static void switch_mind_ki(PlayerType *player_ptr, const PLAYER_LEVEL plev, cons
...
+        return std::string();
     }
 }
 
 /*!
  * @brief 特殊技能の効果情報をまとめたフォーマットを返す
- * @param p 情報を返す文字列参照ポインタ
  * @param use_mind 職業毎の特殊技能ID
  * @param power モンスター魔法のID
+ * @return std::string 電力に関する印刷可能なコメント付き
```
Comment (habu1010) [inline src/mind/mind-info.cpp:141]: The natural Japanese expression is "特殊技能の情報を表す文字列".

Comment (backwardsEric) [reply src/mind/mind-info.cpp]: The forced push should have changed the machine translated result in the comment to your suggestion.

**Comment by backwardsEric** [comment]
The second forced push adds const to the declarations at mind/mind-power-getter.cpp:258 and window/display-sub-window-spells.cpp:125.

**Review by Hourier — APPROVED** [review]
Thank you for reflecting our comments.
I checked the current codes and found no problem.
I approve this.

**Pull request merged**